packages/amplify-category-api/src/provider-utils/awscloudformation/service-walkthroughs/appSync-walkthrough.ts

```diff
@@ -330,111 +479,122 @@ export const updateWalkthrough = async (context): Promise<UpdateApiRequest> => {
   }
 
-  return { authConfig, resolverConfig };
+  context.print.info('');
 }
 
-async function askResolverConflictQuestion(context, modelTypes?) {
-  let resolverConfig: any = {};
+async function displayAuthMode(context, resource, authMode) {
+  if (authMode == 'API_KEY' && resource.output.GraphQLAPIKeyOutput) {
+    let { apiKeys } = await context.amplify.executeProviderUtils(context, 'awscloudformation', 'getAppSyncApiKeys', {
+      apiId: resource.output.GraphQLAPIIdOutput,
+    });
+    let apiKeyExpires = apiKeys.find(key => key.id == resource.output.GraphQLAPIKeyOutput)?.expires;
+    if (!apiKeyExpires) {
+      return authProviderChoices.find(choice => choice.value === authMode).name;
+    }
+    let apiKeyExpiresDate = new Date(apiKeyExpires * 1000);
+    return `${authProviderChoices.find(choice => choice.value === authMode).name} expiring ${apiKeyExpiresDate}: ${resource.output.GraphQLAPIKeyOutput}`;
+  }
+  return authProviderChoices.find(choice => choice.value === authMode).name;
+}
 
-  if (await context.prompt.confirm('Enable conflict detection?')) {
-    const askConflictResolutionStrategy = async msg => {
-      let conflictResolutionStrategy;
-
-      do {
-        const conflictResolutionQuestion: ListQuestion = {
-          type: 'list',
-          name: 'conflictResolutionStrategy',
-          message: msg,
-          default: 'AUTOMERGE',
-          choices: [
-            {
-              name: 'Auto Merge',
-              value: 'AUTOMERGE',
-            },
-            {
-              name: 'Optimistic Concurrency',
-              value: 'OPTIMISTIC_CONCURRENCY',
-            },
-            {
-              name: 'Custom Lambda',
-              value: 'LAMBDA',
-            },
-            {
-              name: 'Learn More',
-              value: 'Learn More',
-            },
-          ],
-        };
-        if (conflictResolutionStrategy === 'Learn More') {
-          conflictResolutionQuestion.prefix = dataStoreLearnMore;
-        }
-        ({ conflictResolutionStrategy } = await inquirer.prompt([conflictResolutionQuestion]));
-      } while (conflictResolutionStrategy === 'Learn More');
+async function askAdditionalQuestions(context, authConfig, defaultAuthType, modelTypes?) {
+  authConfig = await askAdditionalAuthQuestions(context, authConfig, defaultAuthType);
+  return { authConfig };
+}
 
-      let syncConfig: any = {
-        ConflictHandler: conflictResolutionStrategy,
-        ConflictDetection: 'VERSION',
-      };
+async function askResolverConflictQuestion(context, resolverConfig, modelTypes?) {
+  let resolverConfigResponse: any = {};
 
-      if (conflictResolutionStrategy === 'LAMBDA') {
-        const { newFunction, lambdaFunctionName } = await askSyncFunctionQuestion(context);
-        syncConfig.LambdaConflictHandler = {
-          name: lambdaFunctionName,
-          new: newFunction,
-        };
+  if (await context.prompt.confirm('Enable conflict detection?', !resolverConfig?.project)) {
```

Comment:
Should we be using `amplify-prompts` here?

Reply:
These are existing prompts, I haven't planned to migrate these to use amplify-prompts in this PR. We need to have a separate work item.